packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts

```diff
@@ -36,9 +36,19 @@ export class StackMetadataBackendOutputStorageStrategy
       new CfnOutput(this.stack, key, { value });
     });
 
+    const metadataValue =
+      this.stack.templateOptions.metadata &&
+      this.stack.templateOptions.metadata[keyName]
+        ? this.stack.templateOptions.metadata[keyName]
+        : { stackOutputs: [] };
+
     this.stack.addMetadata(keyName, {
+      ...metadataValue,
       version: backendOutputEntry.version,
-      stackOutputs: Object.keys(backendOutputEntry.payload),
+      stackOutputs: [
+        ...metadataValue.stackOutputs,
+        ...Object.keys(backendOutputEntry.payload),
+      ],
```

Comment:
Could you explain this change?

Edit: after reading rest of the code I'm guessing this is to solve a problem of "multiple defineStorage contributing to same stack metadata/output key". 
I was facing similar problem in the POC I was building for custom client config. This wasn't merged as final approach was different but you might find it useful for this PR, see https://github.com/aws-amplify/amplify-backend/pull/946#discussion_r1466988928 (i.e. `addBackendOutputEntry` and logic around it).

Reply:
Could we maybe use `appendToBackendOutputList` (see [here](https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts#L48)) to lazily append multiple buckets to the same output key similar to what is done for functions? See [here](https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-function/src/factory.ts#L390).

Reply:
Good point @rtpascual! I somehow didn't see `appendToBackendOutputList` in this file.

Reply:
Done [e8548cf](https://github.com/aws-amplify/amplify-backend/pull/1742/commits/e8548cf39936a5a284ce458e2390f7eb01cc833e)